esp-qcom-image: convert to OE-core uki.bbclass

koenkooi opened the pull request:
This new class uses the OE-core uki.bbclass infrastructure to assemble an
ESP image with systemd-boot + UKI. It uses method #2 for config-less booting.

Systemd-boot is installed through package management, the UKI is picked
up from deploy. 

Signed-off-by: Koen Kooi [koen.kooi@oss.qualcomm.com](mailto:koen.kooi@oss.qualcomm.com)
Fixes https://github.com/qualcomm-linux/meta-qcom/issues/707

koenkooi commented:
See https://github.com/qualcomm-linux/meta-qcom-hwe/pull/138 for review and comment history

ndechesne commented on classes/uki-esp-image.bbclass:
```diff
@@ -0,0 +1,16 @@
+
+# Optional subfolder, dependant on where the ESP partition gets mounted
+# intended to only have a leading slash, no trailing slash e.g. '/EFI', or just empty, ''
+ESPFOLDER ?= "/EFI"
```
in which case/scenario are we expected to use ESPFOLDER? is that really needed?

lumag replied:
It should be  somthing like `/boot/efi` by default

koenkooi replied:
For non-qcom systems the ESP can be mounted in different places, e.g. /boot/efi/, in our case it's mounted directly as /boot, hence "EFI" being in the root.

ricardosalveti replied:
Right, this is just for separating the content in a different folder, not related to how it gets mounted later (which is done by systemd automatically).

ndechesne commented on recipes-kernel/images/esp-qcom-image.bb:
```diff
@@ -1,14 +1,25 @@
+
+UKI_CMDLINE="root=${QCOM_BOOTIMG_ROOTFS} rw rootwait"
+
+# Remove leading 'qcom/' from KERNEL_DEVICEREE, we store the file in DEPLOY_DIR_IMAGE directly, not in the qcom/ subfolder
```
do we need to keep these 2 lines here?

koenkooi replied:
The bottom one can go for sure, the top on is still needed AFAICT.

ndechesne reviewed (commented):
looks neat!

lumag commented:
@koenkooi I think this is more or less ready now. Could you please squash those commits? Let's merge it, unless there are any obstacles.

lumag commented:
@koenkooi ping, let's merge this.

koenkooi commented:
> @koenkooi ping, let's finish and merge this.

I've squashed and rebased it

lumag commented:
> > @koenkooi ping, let's finish and merge this.
> 
> I've squashed and rebased it

I think it's now failing on the copyright checks. LGTM otherwise.

koenkooi commented:
> > > @koenkooi ping, let's finish and merge this.
> > 
> > 
> > I've squashed and rebased it
> 
> I think it's now failing on the copyright checks. LGTM otherwise.

I've added a copyright header, but I do require some feedback on that:

1. I've used 2025 since it's a new class, but the recipe it is based on is much older, is 2025 the right year to use?
2. I want metadata to be MIT, the original recipe is 3BSD, does this fit into the meta-qcom licensing policy?

ricardosalveti commented on classes/uki-esp-image.bbclass:
```diff
@@ -0,0 +1,16 @@
+
+addtask ukiesp after do_deploy uki before do_image_complete do_image_wic
+
+
```
Nit, drop these extra 2 empty lines.

koenkooi replied:
Done

ricardosalveti commented on recipes-kernel/images/esp-qcom-image.bb:
```diff
@@ -1,14 +1,22 @@
+inherit uki
+UKI_FILENAME = "${EFI_LINUX_IMG}"
+
+UKI_CMDLINE="root=${QCOM_BOOTIMG_ROOTFS} rw rootwait"
```
Nit, space between '='.

koenkooi replied:
Done

ricardosalveti commented on recipes-kernel/images/esp-qcom-image.bb:
```diff
@@ -1,14 +1,22 @@
 
-inherit image
+ESPFOLDER = ""
+inherit image uki-esp-image
```
Why not add uki here as well?

koenkooi replied:
Done

ricardosalveti commented:
> 1. I've used 2025 since it's a new class, but the recipe it is based on is much older, is 2025 the right year to use?

You could define the range, but if owned by Qualcomm just updating to the latest is fine.

> 2. I want metadata to be MIT, the original recipe is 3BSD, does this fit into the meta-qcom licensing policy?

Yes, MIT is fine, that is our main license here.

ricardosalveti commented:
Jobs failed to run https://github.com/qualcomm-linux/meta-qcom/actions/runs/14358395851

ricardosalveti commented:
Should had removed my approval, this PR is not yet fully functional, as I pasted above it fails to boot now.
